File: Kotlin/프로그래머스/1/140108. 문자열 나누기/문자열 나누기.kt
class Solution {
    fun solution(s: String): Int {
        var answer: Int = 0

        val sChar = s.toCharArray()

        var wait = 0
        for (i in sChar.indices) {
            if (i < wait) {
                continue
            }
            val c = sChar[i]

            var cCount = 0
            var otherCount = 0

            for (j in i until sChar.size) {

                if (sChar[j] == c) {
                    cCount++
                } else {
                    otherCount++
                }

                if (cCount == otherCount || j == sChar.size - 1) {
                    answer++
                    wait = j + 1
                    break
                }
            }
        }

        return answer
    }
}

File: C#/프로그래머스/1/140108. 문자열 나누기/문자열 나누기.cs
using System;

public class Solution {
    public int solution(string s) {
        int answer = 0;
        int same = 0;
        char f = s[0];
        
        for(int i = 0; i < s.Length - 1; i++)
        {
            char c = s[i];
            if(c == f) 
            {
                same++;
                continue;
            }
        
            same--;
            if(same == 0) 
            {
                answer++;
                f = s[i+1];
            }
            
        }
        
        return answer + 1;
    }
}